Examples/Scripts/FreeSurferPipelineBatch-long.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash 
+
+get_usage_and_exit(){
+    echo "usage: "
+    echo "FreeSurferPipelineBatch-long.sh [options]"
+    echo "options:"
+    echo "  --runlocal                      run locally [FALSE]"
+    exit -1
+}
+
+command_line_specified_run_local=FALSE
+while [ -n "$1" ]; do
+    case "$1" in
+        --runlocal) shift; command_line_specified_run_local=TRUE ;;
+        *) shift ;;
+    esac
+done
+
+#################################################################################################
+# General input variables
+##################################################################################################
+#Location of Subject folders (named by subjectID)
+StudyFolder="/media/myelin/brainmappers/Connectome_Project/MishaLongitudinal/hcp/FSLong-test" 
+#list of subject labels, space separated
+Subjlist="HCA6002236"
+#list of timepoints per each subject. Subject timepoint groups are space separated, timepoints within subject are @ separated
+Timepoint_list="HCA6002236_V1_MR@HCA6002236_V2_MR"
+#list of longitudinal template labels, one per subject, space separated
+Template_list="HCA6002236_V1_V2"
+
+EnvironmentScript="/media/myelin/brainmappers/Connectome_Project/MishaLongitudinal/hcp/scripts/SetUpHCPPipeline-long.sh" #Pipeline environment script
+
+# Requirements for this script
+#  installed versions of: FSL, FreeSurfer, Connectome Workbench (wb_command), gradunwarp (HCP version)
+#  environment: HCPPIPEDIR, FSLDIR, FREESURFER_HOME, CARET7DIR, PATH for gradient_unwarp.py
+
+# If you want to use FreeSurfer 5.3, change the ${queuing_command} line below to use
+# ${HCPPIPEDIR}/FreeSurfer/FreeSurferPipeline-v5.3.0-HCP.sh
+
+#Set up pipeline environment variables and software
+source "$EnvironmentScript"
+
+# Log the originating call
+echo "$@"
+
+#NOTE: syntax for QUEUE has changed compared to earlier pipeline releases,
+#DO NOT include "-q " at the beginning
+#default to no queue, implying run local
+QUEUE=""
+#QUEUE="hcp_priority.q"
+
+########################################## INPUTS ########################################## 
+
+#Scripts called by this script do assume they run on the outputs of the PreFreeSurfer Pipeline
+
+######################################### DO WORK ##########################################
+
+Template_list=( ${Template_list[@]} )
```

Comment:
You could do this instead above:
```
Subjlist=(123456 654321)
Timepoint_list=(123456_V1@123456_V2 654321_V1@654321_V2)
```
And then drop these lines.  I don't know how others feel about putting user-edited lists into array parenthesis instead of quotes, though.

It doesn't make a lot of sense to do `var=(${var[@]})` when var is a string, `var=($var)` would work.

As for the logic, I wonder if we should instead expect a consistent naming convention, and just have a "here are the generic possible visit identifiers", and then build the folder names and test if they exist:

```
Subjlist="123456 654321"
PossibleVisits="V1 V2 V3"
#in case we want to exclude some without deleting them
BadVisitList="132465_V2 645312_V3"
...
for subject in $Subjlist
do
    TemplateName="$subject"
    for visit in $PossibleVisits
    do
        badvisit=0
        for check in $BadVisitList
        do
            if [[ "$check" == "$subject"_"$visit" ]]; then badvisit=1; fi
        done
        if ((badvisit)); then continue; fi

        visitfolder="$StudyFolder"/"$subject"_"$visit"_MR
        if [[ -d "$visitfolder" ]]
        then
            TemplateName+=_"$visit"
...
```
This would make it less tedious for users to edit the example to fit their data.  If we wouldn't have visits with bad structurals in the StudyFolder, we wouldn't need the exclude code either.

Reply:
I like this idea as long as it is a batch script thing rather than a pipeline script thing.

Reply:
> I don't know how others feel about putting user-edited lists into array parenthesis instead of quotes, though
I used quotes because that seems to be standard across batch scripts. 

> I wonder if we should instead expect a consistent naming convention, and just have a "here are the generic possible visit identifiers", and then build the folder names and test if they exist.
I agree with Matt that this can be a batch script option.

Reply:
I incorporated needed logic into FreeSurferPipelineBatch-long.sh and PostFreeSurferPipelineBatch-long.sh.